pkg/inject/inject.go

```diff
@@ -342,8 +348,9 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.pod.meta = &v.ObjectMeta
 
 		if conf.ownerRetriever != nil {
-			kind, name := conf.ownerRetriever(v)
-			switch kind {
+			conf.workload.ownerRef = conf.ownerRetriever(v)
+			name := conf.workload.ownerRef.Name
+			switch strings.ToLower(conf.workload.ownerRef.Kind) {
```

Comment:
non-actionable feedback (unrelated to this PR) to think about going forward: all of the `strings.ToLower` to match `Kind` to our `k8s.*` constants isn't ideal.

Reply:
Yeah, I'll see if I can see how they handle that in the k8s source code.